diesel/src/query_dsl/join_dsl.rs

```diff
@@ -44,6 +44,84 @@ where
     }
 }
 
+/// Methods allowing various joins between two or more tables.
+///
+/// If you have invoked [`joinable!`] for the two tables, you can pass that
+/// table directly.  Otherwise you will need to use [`.on`] to specify the `ON`
+/// clause.
+///
+/// [`joinable!`]: ../macro.joinable.html
+/// [`.on`]: trait.JoinOnDsl.html#method.on
+///
+/// You can join to as many tables as you'd like in a query, with the
+/// restriction that no table can appear in the query more than once. The reason
+/// for this restriction is that one of the appearances would require aliasing,
+/// and we do not currently have a fleshed out story for dealing with table
+/// aliases.
+///
+/// You may also need to call [`allow_tables_to_appear_in_same_query!`][] (particularly if
+/// you see an unexpected error about `AppearsInFromClause`). See the
+/// documentation for [`allow_tables_to_appear_in_same_query!`][] for details.
+///
+/// Diesel expects multi-table joins to be semantically grouped based on the
+/// relationships. For example, `users.inner_join(posts.inner_join(comments))`
+/// is not the same as `users.inner_join(posts).inner_join(comments)`. The first
+/// would deserialize into `(User, (Post, Comment))` and generate the following
```

Comment:
Is there an example for this? I saw `(User, Post, Comment)` but I don't know why it isn't working, it would be good to have example so easier to follow. The example below uses explicit select.

Reply:
@pickfire Please don't ask questions on old PR's. That's definitively not the right place for such things. Use the gitter channel or our forum instead.